packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1417,6 +1417,83 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     return child;
   }
 
+  /// The [DelegatedTransition] provided to the route below this one in the
+  /// navigation stack.
+  ///
+  /// {@macro flutter.widgets.delegatedTransition}
+  ///
+  /// The [ModalRoute] receiving this transition will set it to their
+  /// [receivedTransition] property.
+  DelegatedTransition? get delegatedTransition => null;
+
+  /// The [DelegatedTransition] received from the route above this one in the
+  /// navigation stack.
+  ///
+  /// {@macro flutter.widgets.delegatedTransition}
+  ///
+  /// `receivedTransition` will use the above route's [delegatedTransition] in
+  /// order to play the right route transition when the above route either enters
```

Comment:
Nit: "play" => "use"? I'm just worrying that the terminology "play"ing a transition hasn't been used before in our docs maybe.